test/plugin/base_test.rb

```diff
@@ -605,6 +605,54 @@ def test_sanitize_tags_from_container_name_with_require_valid_tags_false
     end
   end
 
+  def test_configure_split_logs_by_tag
+    setup_gce_metadata_stubs
+    {
+      APPLICATION_DEFAULT_CONFIG => true,
+      DISABLE_SPLIT_LOGS_BY_TAG_CONFIG => false
+    }.each do |(config, split_logs_by_tag)|
+      d = create_driver(config)
+      assert_equal split_logs_by_tag,
+                   d.instance.instance_variable_get(:@split_logs_by_tag)
+    end
+  end
+
+  def test_split_logs_by_tag
+    setup_gce_metadata_stubs
+    log_entries_count = 5
+    [
+      [true, APPLICATION_DEFAULT_CONFIG, log_entries_count,
+       COMPUTE_PARAMS.merge(log_name: 'multi')],
+      # Request level log name should be '' if should_split is false.
+      [false, DISABLE_SPLIT_LOGS_BY_TAG_CONFIG, 1,
+       COMPUTE_PARAMS.merge(log_name: 'multi', request: { log_name: '' })]
+    ].each do |(should_split, config, requests_count, params)|
+      setup_prometheus
+      setup_logging_stubs do
+        @logs_sent = []
+        d = create_driver(config + PROMETHEUS_ENABLE_CONFIG, 'test', true)
+        log_entries_count.times do |i|
+          d.emit("tag#{i}", 'message' => log_entry(i))
+        end
+        d.run
+        verify_log_entries(log_entries_count, params,
+                           'textPayload') do |entry, i|
+          verify_default_log_entry_text(entry['textPayload'], i, entry)
+          # expected_log_name should be nil if should_split is true.
+          expected_log_name =
+            "projects/test-project-id/logs/tag#{i}" unless should_split
+          assert_equal expected_log_name, entry['logName']
+        end
+        assert_prometheus_metric_value(:stackdriver_successful_requests_count,
+                                       requests_count,
+                                       grpc: use_grpc, code: ok_status_code)
+        assert_prometheus_metric_value(:stackdriver_ingested_entries_count,
+                                       log_entries_count,
+                                       grpc: use_grpc, code: ok_status_code)
+      end
+    end
```

Comment:
You could just add a loop over `@logs_sent` that checks the `log_name` in each request, e.g.:
```ruby
@logs_sent.each_with_index do |log, i|
  expected_log_name =
    "projects/test-project-id/logs/tag#{i}" if should_split
  assert_equal expected_log_name, log['logName']
end
```

Reply:
Should you not still check that the log name in the request vs. the log name in each entry is set correctly?

Reply:
I kind of feel that it's redundant if we are checking the combined log names (`log_name = entry['logName'] || request['logName']`) as well. But since I'm pulling that out of the `verify_log_entries` method in the latest commit, I've restored the checks for request level log names and entry level log names separately.

Reply:
I guess I was suggesting that you check the absence of the log name in places where it wasn't supposed to be, which, together with the combined checks, would have produced the right result. But doing them explicitly works too.

Reply:
Cool cool. Yeah, I was also debating whether to check both explicitly or check the combined. I guess either works.